service/src/main/java/de/starwit/service/impl/PostFlywayService.java

```diff
@@ -0,0 +1,149 @@
+package de.starwit.service.impl;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import de.starwit.persistence.entity.MitigationActionTypeEntity;
+import de.starwit.persistence.entity.TrafficIncidentEntity;
+import de.starwit.persistence.entity.TrafficIncidentTypeEntity;
+import de.starwit.persistence.repository.MitigationActionTypeRepository;
+import de.starwit.persistence.repository.TrafficIncidentRepository;
+import de.starwit.persistence.repository.TrafficIncidentTypeRepository;
+import jakarta.annotation.PostConstruct;
+
+/**
+ * Service class for importing sample data. Will only be executed,
+ * if flag in application properties is set.
+ */
+
+@Service
+public class PostFlywayService {
+
+    static final Logger LOG = LoggerFactory.getLogger(PostFlywayService.class);
+
+    @Value("${scenario.setup:false}")
+    private boolean setupScenario = true;
+
+    @Value("${scenario.importFolder:/scenariodata/traffic/}")
+    private String scenarioImportFolder;
+
+    private String mititgationTypeFileName = "mitigationtypes.json";
+
+    private String incidentTypeFileName = "incidenttypes.json";
+
+    private String demoDataFileName = "demodata.json";
+
+    @Autowired
+    private TrafficIncidentTypeRepository ttRepository;
+
+    @Autowired
+    private MitigationActionTypeRepository mtRepository;
+
+    @Autowired
+    private TrafficIncidentRepository incidentRepository;
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    @PostConstruct
+    public void init() {
+        LOG.info("Importing sample data: " + setupScenario);
+        if (setupScenario) {
+            LOG.info("Importing sample data structure from folder " + scenarioImportFolder);
+            if (importMitigationTypes(scenarioImportFolder)) {
+                LOG.info("Import of mitigation types successful, importing incident types");
+                if (importIncidentTypes(scenarioImportFolder)) {
+                    importDemoData(scenarioImportFolder);
+                }
+            }
+        }
+    }
+
+    private boolean importMitigationTypes(String folder) {
+        if (mtRepository.findAll().size() > 0) {
+            LOG.info("Mitigation types already imported. Skipping import.");
+        } else {
```

Comment:
I would avoid else and return immediately

Reply:
done